src/Extensions/Diagnostics.HealthChecks/ServiceCollectionExtensions.cs

```diff
@@ -45,5 +62,30 @@ public static IServiceCollection AddHealthCheckPublisher<TPublisher>(this IServi
 			serviceCollection.TryAddEnumerable(ServiceDescriptor.Singleton<IHealthCheckPublisher, TPublisher>());
 			return serviceCollection;
 		}
+
+		/// <summary>
+		/// 
+		/// </summary>
+		/// <param name="serviceCollection"></param>
+		/// <returns></returns>
+		public static IServiceCollection AddRestHealthCheckPublisher(this IServiceCollection serviceCollection)
+		{
+			serviceCollection.TryAddEnumerable(ServiceDescriptor.Singleton<IHealthCheckPublisher, RestHealthCheckPublisher>(
+				(serviceProvider) =>
+				{
+					return new RestHealthCheckPublisher(
+						clusterEndpoints: new Uri(serviceProvider
+													.GetRequiredService<IOptions<RestHealthCheckPublisherOptions>>()
+													.Value
+													.RestHealthPublisherClusterEndpoint),
+						serviceId: serviceProvider
+													.GetRequiredService<IOptions<RestHealthCheckPublisherOptions>>()
+													.Value
+													.RestHealthPublisherServiceId
+					);
+				}
```

Comment:
I think it would be better to get `IOptions<RestHealthCheckPublisherOptions>` as a constructor parameter for `RestHealthCheckPublisher` and then it could be resolved as a regular type without an explicit factory.